pandas/core/generic.py

```diff
@@ -8518,24 +8518,198 @@ def _doc_parms(cls):
-axis : %(axis_descr)s
+axis : {0 or 'index', 1 or 'columns'}, default 0
 skipna : boolean, default True
     Exclude NA/null values. If an entire row/column is NA, the result
-    will be NA
+    will be NA.
+*args : default None
+**kwargs : default None
+    Additional keywords have no effect but might be accepted for
+    compatibility with NumPy.
 
 Returns
 -------
-%(outname)s : %(name1)s\n
-
-
+%(outname)s : %(name1)s or %(name2)s\n
+%(examples)s
 See also
 --------
-pandas.core.window.Expanding.%(accum_func_name)s : Similar functionality
+core.window.Expanding.%(accum_func_name)s : Similar functionality
     but ignores ``NaN`` values.
+Series.%(outname)s : Return %(desc)s over Series axis.
+DataFrame.cummax : Return cumulative maximum over DataFrame axis.
+DataFrame.cummin : Return cumulative minimum over DataFrame axis.
+DataFrame.cumsum : Return cumulative sum over DataFrame axis.
+DataFrame.cumprod : Return cumulative product over DataFrame axis.
+"""
 
+_cummax_examples = """\
+Examples
+--------
+**DataFrame**
+
+>>> df = pd.DataFrame([[7, 1],
+...                    [3, 4],
+...                    [8, 0]],
+...                    columns=list('AB'))
+>>> df
+   A  B
+0  7  1
+1  3  4
+2  8  0
+
+**axis**
+
+axis=None : Iterates over rows and finds the cumulative value in each column.
+If value is different from the previous one, updates it:
+
+>>> df.cummax(axis=None)
+   A  B
+0  7  1
+1  7  4
+2  8  4
+>>> df.cummin(axis=None)
+   A  B
+0  7  1
+1  3  1
+2  3  0
+>>> df.cumsum(axis=None)
+    A  B
+0   7  1
+1  10  5
+2  18  5
+>>> df.cumprod(axis=None)
+     A  B
+0    7  1
+1   21  4
+2  168  0
+
+axis=1 : Iterates over columns and finds the cumulative value in each row.
+If value is different from the previous one, updates it:
+
+>>> df.cummax(axis=1)
+   A  B
+0  7  7
+1  3  4
+2  8  8
+>>> df.cummin(axis=1)
+   A  B
+0  7  1
+1  3  3
+2  8  0
+>>> df.cumsum(axis=1)
+   A  B
+0  7  8
+1  3  7
+2  8  8
+>>> df.cumprod(axis=1)
+   A   B
+0  7   7
+1  3  12
+2  8   0
+
+**skipna**
+
+skipna=True : Ignores NaN values during operation:
+
+>>> df = pd.DataFrame([[7, np.nan],
+...                    [np.nan, 4],
+...                    [8, 0]],
+...                    columns=list('AB'))
+>>> df
+     A    B
+0  7.0  NaN
+1  NaN  4.0
+2  8.0  0.0
+
+>>> df.cummax(skipna=True)
+     A    B
+0  7.0  NaN
+1  NaN  4.0
+2  8.0  4.0
+>>> df.cummin(skipna=True)
+     A    B
+0  7.0  NaN
+1  NaN  4.0
+2  7.0  0.0
+>>> df.cumsum(skipna=True)
+      A    B
+0   7.0  NaN
+1   NaN  4.0
+2  15.0  4.0
+>>> df.cumprod(skipna=True)
+      A    B
+0   7.0  NaN
+1   NaN  4.0
+2  56.0  0.0
+
+skipna=False : Includes NaN values:
+
+>>> df.cummax(skipna=False)
+     A   B
+0  7.0 NaN
+1  NaN NaN
+2  NaN NaN
+>>> df.cummin(skipna=False)
+     A   B
+0  7.0 NaN
+1  NaN NaN
+2  NaN NaN
+>>> df.cumsum(skipna=False)
+     A   B
+0  7.0 NaN
+1  NaN NaN
+2  NaN NaN
+>>> df.cumprod(skipna=False)
+     A   B
+0  7.0 NaN
+1  NaN NaN
+2  NaN NaN
+
+**Series**
+
```

Comment:
If we decide to have separate string examples for each method, we can keep the examples for `Series`.

Reply:
> If we decide to have separate string examples for each method, we can keep the examples for Series.

\+ 1

Another suggestion would be to start with Series to just illustrate the concept of "cumulative max", as this will make the examples a little bit easier, and show the effect of NaNs. And then show DataFrame, saying that by default the same happens for each column of the DataFrame, and optionally use `axis=1` to take cumulative max for each row.

Reply:
Thanks for the suggestion. I agree, it is definitely easier to see what is going on with NaNs if we use a Series example instead of DataFrame. I will change it this way.